compiler/src/dotty/tools/dotc/core/CheckRealizable.scala

```diff
@@ -89,18 +95,36 @@ class CheckRealizable(implicit ctx: Context) {
       else boundsRealizability(tp).andAlso(memberRealizability(tp))
   }
 
-  /** `Realizable` if `tp` has good bounds, a `HasProblemBounds` instance
-   *  pointing to a bad bounds member otherwise.
+  /** `Realizable` if `tp` has good bounds, a `HasProblem...` instance
+   *  pointing to a bad bounds member otherwise. "Has good bounds" means:
+   *
+   *    - all type members have good bounds
+   *    - all base types are class types, and if their arguments are wildcards
+   *      they have good bounds.
    */
   private def boundsRealizability(tp: Type) = {
-    def hasBadBounds(mbr: SingleDenotation) = {
-      val bounds = mbr.info.bounds
-      !(bounds.lo <:< bounds.hi)
-    }
-    tp.nonClassTypeMembers.find(hasBadBounds) match {
-      case Some(mbr) => new HasProblemBounds(mbr)
-      case _ => Realizable
+    val mbrProblems =
+      for {
+        mbr <- tp.nonClassTypeMembers
+        if !(mbr.info.loBound <:< mbr.info.hiBound)
+      }
+      yield new HasProblemBounds(mbr)
+      
+    def baseTypeProblems(base: Type) = base match {
+      case AndType(base1, base2) =>
```

Comment:
Complication: Realizablility used to just check that all type members have good bounds. Now it has to look at basetypes as well.